### PR TITLE
[Feature] Parallel environments lazy heterogenous data compatibility 

### PR DESCRIPTION
**Pull request by matteobettini**

### CODE REVIEW

File: test/test_env.py [hunk 1]
```diff
@@ -1946,6 +1946,32 @@ def test_rollout_policy(self, batch_size, rollout_steps, count):
             else:
                 assert (td["lazy"][..., i]["action"] == 0).all()
 
+    @pytest.mark.parametrize("batch_size", [(), (1, 2), (2,)])
+    @pytest.mark.parametrize("env_type", ["serial", "parallel"])
+    def test_vec_env(self, batch_size, env_type, rollout_steps=4, n_workers=2):
```
Comment (vmoens) [inline test/test_env.py:1951]: each parallel test takes a long time to launch. Are all the combinations absolutely necessary? Can we have just one Parallel?

File: test/test_env.py [hunk 2]
```diff
@@ -1946,6 +1946,32 @@ def test_rollout_policy(self, batch_size, rollout_steps, count):
...
+        env_fun = lambda: HeteroCountingEnv(batch_size=batch_size)
+        if env_type == "serial":
+            vec_env = SerialEnv(n_workers, env_fun)
+        else:
+            vec_env = ParallelEnv(n_workers, env_fun)
+        vec_batch_size = (n_workers,) + batch_size
+        # check_env_specs(vec_env, return_contiguous=False)
+        policy = HeteroCountingEnvPolicy(vec_env.input_spec["_action_spec"])
```
Comment (vmoens) [inline test/test_env.py:1959]: Note to self: `input_spec["_action_spec"]` is clunky, we should have some other way to access the composite spec of the action.

File: torchrl/data/utils.py [hunk 3]
```diff
@@ -180,6 +180,22 @@ def check_no_exclusive_keys(spec: TensorSpec, recurse: bool = True):
     return True
 
 
+def contains_lazy_spec(spec: TensorSpec) -> bool:
```
Comment (vmoens) [inline torchrl/data/utils.py:183]: add to doc

**Review by vmoens — APPROVED** [review]
LGTM some minor comments

File: test/test_env.py (hunk 2 above)
Comment (matteobettini) [reply test/test_env.py:1959]: +1

**Pull request merged**